### PR TITLE
Minor fixup for RPCS3 termination

### PR DESCRIPTION
**Pull request by Nekotekina**
@Megamouse reported some threads possibly stuck on exit.

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -117,6 +122,7 @@ void Emulator::Init()
 
 	idm::init();
 	g_fxo->reset();
+	g_fxo->init<named_thread<progress_dialog_server>>();
```
Comment (elad335) [inline rpcs3/Emu/System.cpp:125]: In RSX replay as well?

File: rpcs3/util/logs.cpp [hunk 2]
```diff
@@ -412,7 +412,7 @@ logs::file_writer::file_writer(const std::string& name, u64 max_size)
 	{
 		thread_ctrl::set_native_priority(-1);
 
-		while (true)
+		while (m_out != umax)
```
Comment (Megamouse) [inline rpcs3/util/logs.cpp:415]: I don't think this change is correct.

But I found out that this causes file_writer::flush to endlessly return false:
`if (end > st) // end == st == 363895`

363895 is obviously way smaller than umax

**Comment by Megamouse** [comment]
Also, Progress Dialog Server is still not terminated

**Comment by Megamouse** [comment]
The entire issue is resolved if I quit the application with a delay after Emu.stop().
So maybe there's a specific thread that we to wait for after stopping?

File: rpcs3/util/logs.cpp (hunk 2 above)
Comment (elad335) [reply rpcs3/util/logs.cpp:415]: this change changes nothing actually.
need to check if the destructor is called at all or m_out signaling gets discarded.

**Pull request merged**